app/facades/trip_facade.rb

```diff
@@ -46,7 +47,7 @@ def destination_data
   end
 
   def leg_data
-    @_leg_data ||= LegInfo.new(_directions.leg_info)
+    LegSerializer.from_json(_directions.leg_info, @trip.id, 1)
```

Comment:
This can probably be refactored out given the serialization you set up in `TripLegs` Created #73